_schemas/1.0/schema.json

```diff
@@ -196,202 +202,293 @@
         },
-        "source": {
-          "type": "object",
-          "properties": {
-            "pointer": {
-              "description": "A JSON Pointer [RFC6901] to the associated entity in the request document [e.g. \"/data\" for a primary data object, or \"/data/attributes/title\" for a specific attribute].",
-              "type": "string"
-            },
-            "parameter": {
-              "description": "A string indicating which query parameter caused the error.",
-              "type": "string"
-            }
-          }
+        "relationships": {
+          "$ref": "#/definitions/relationships"
+        }
+      },
+      "additionalProperties": false
+    },
+    "resourceCollection": {
+      "description": "An array of resource objects.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/resource"
+      },
+      "uniqueItems": true
+    },
+    "resourceIdentification": {
+      "type": "object",
+      "required": [
+        "type",
+        "id"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "pattern": "^[a-zA-Z0-9]{1}(?:[-\\w]*[a-zA-Z0-9])?$"
         },
+        "id": {
+          "type": "string"
+        }
+      }
+    },
+    "resourceIdentifier": {
+      "description": "The \"type\" and \"id\" to non-empty members.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/resourceIdentification"
+        }
+      ],
+      "properties": {
+        "id": true,
+        "type": true,
         "meta": {
           "$ref": "#/definitions/meta"
         }
       },
       "additionalProperties": false
+    },
+    "resourceLinks": {
+      "description": "The top-level links object **MAY** contain the following members: self, related, pagination links.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/links"
+        }
+      ],
+      "properties": {
+        "self": {
+          "description": "",
+          "$ref": "#/definitions/link"
+        }
+      },
+      "additionalProperties": false
+    },
+    "topLevelLinks": {
+      "description": "The top-level links object **MAY** contain the following members: self, related, pagination links.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/links"
+        },
+        {
+          "description": "Pagination links for the primary data.",
+          "$ref": "#/definitions/pagination"
+        }
+      ],
+      "properties": {
+        "self": {
+          "description": "The link that generated the current response document.",
+          "$ref": "#/definitions/link"
+        },
+        "related": {
+          "description": "A related resource link when the primary data represents a resource relationship.",
+          "$ref": "#/definitions/link"
+        },
+        "first": true,
+        "last": true,
+        "next": true,
+        "prev": true
+      },
+      "additionalProperties": false
```

Comment:
I believe this is incorrect. The spec gives various examples of additional link types, and the 1.1 spec (though I know that's not what this covers) includes examples such as `describedby`.

Reply:
To be clear, I'm suggesting removing the `additionalProperties: false` rule.

Reply:
Ref for `describedby` - https://github.com/json-api/json-api/pull/1447

This makes me question myself on whether or not the 1.0 spec specifically does or does not preclude other link types. It seems vague, and doesn't use any `MUST` or `MUST NOT` language, so I think `describedby` should be valid on 1.0, but it's... vague.

Reply:
Ack - I am actually wrong. Per the spec,

```
Unless otherwise noted, objects defined by this specification MUST NOT contain any additional members. Client and server implementations MUST ignore members not recognized by this specification.
```

Sooo this is yet another reason to really get 1.1 published, since implementations like Drupal's `jsonapi_hypermedia` add `describedby`.

Reply:
This is exactly the reason why i added the `additionalProperties: false` rule. I was not sure the links object `MUST` or `MUST NOT` contain any additional members. So, when in doubt, I preferred to refer to :
`Unless otherwise noted, objects defined by this specification MUST NOT contain any additional members. Client and server implementations MUST ignore members not recognized by this specification.`

Reply:
Yep. That's the correct interpretation. I don't have permission to resolve this thread but feel free to do so. Thanks for your awesome work on this.